pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -1817,14 +1817,57 @@ def set_mcp_sampling_model(self, model: models.Model | models.KnownModelName | s
         except exceptions.UserError as e:
             raise exceptions.UserError('No sampling model provided and no model set on the agent.') from e
 
-        from .mcp import MCPServer
-
         def _set_sampling_model(toolset: AbstractToolset[AgentDepsT]) -> None:
             if isinstance(toolset, MCPServer):
                 toolset.sampling_model = sampling_model
 
         self._get_toolset().apply(_set_sampling_model)
 
+    def set_mcp_elicitation_toolset(self, toolset_for_elicitation: AbstractToolset[Any] | None = None) -> None:
```

Comment:
On second thought, I think we should drop this method as it's too mcp-run-python specific to live on agent. I think it's fine to require the user to pass their own toolset to create_tool_elicitation_callback (and create_auto_tool_injection_callback)